service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -278,4 +295,42 @@ void getDatasetPreviewTable() {
         datasetPreviewTable.getColumns().get(0),
         is(new bio.terra.catalog.model.ColumnModel().name("column a").arrayOf(false)));
   }
+
+  @Test
+  void testExportDatarepoDataset() {
+    reset(datasetDao);
+    when(datasetDao.retrieve(datasetId)).thenReturn(tdrDataset);
+    when(datarepoService.getRole(user, sourceId)).thenReturn(DatasetAccessLevel.READER);
+    String workspaceId = String.valueOf(UUID.randomUUID());
+    doThrow(new BadRequestException("error"))
+        .when(datarepoService)
+        .exportDatarepoDataset(user, sourceId, workspaceId);
+    assertThrows(
+        BadRequestException.class,
+        () -> datasetService.exportDataset(user, datasetId, workspaceId));
+  }
+
+  @Test
+  void testExportExternalDataset() {
+    reset(datasetDao);
+    when(datasetDao.retrieve(datasetId)).thenReturn(externalDataset);
+    String workspaceId = String.valueOf(UUID.randomUUID());
+    assertThrows(
+        UnauthorizedException.class,
+        () -> datasetService.exportDataset(user, datasetId, workspaceId));
+  }
+
+  @Test
+  void testExportWorkspaceDataset() {
+    reset(datasetDao);
+    when(datasetDao.retrieve(datasetId)).thenReturn(workspaceDataset);
+    when(rawlsService.getRole(user, workspaceSourceId)).thenReturn(DatasetAccessLevel.READER);
+    String workspaceId = String.valueOf(UUID.randomUUID());
+    datasetService.exportDataset(user, datasetId, workspaceId);
+
+    DatasetService mockDatasetService = mock(DatasetService.class);
+    doNothing().when(mockDatasetService).exportDataset(user, datasetId, workspaceId);
+    mockDatasetService.exportDataset(user, datasetId, workspaceId);
+    verify(mockDatasetService).exportDataset(user, datasetId, workspaceId);
```

Comment:
I don't think this is necessary with strict subbing. If it's not called, you should get a test failure because of the mock on line 332.